apps/xprof_gui/priv/src/components/monitoring/GraphPanel/GraphPanel.jsx

```diff
@@ -35,27 +44,39 @@ const GraphPanel = ({
   expand,
   shrink,
   calleeClick,
-}) => (
-  <div className="panel panel-default">
-    <GraphPanelHeading
-      mfa={mfa}
-      stopMonitoringFunction={() => stopMonitoringFunction(mfa)}
-      callees={callees}
-      calleesVisibility={calleesVisibility}
-      showCallees={() => showCallees(mfa[3])}
-      hideCallees={() => hideCallees(mfa[3])}
-      panelVisibility={panelVisibility}
-      expand={() => expand(mfa[3])}
-      shrink={() => shrink(mfa[3])}
-      calleeClick={calleeClick}
-    />
-    {panelVisibility ? (
-      <div className="panel-body">
-        <Graph dps={dps} />
-      </div>
-    ) : null}
-  </div>
-);
+  setSize,
+  size,
+}) => {
+  const monitoredID = safecomposeID(monitored.query);
```

Comment:
Move `monitoredID` to the state. Dispatch an action in `getMonitoredFunctions` which will create/delete IDs.